packages/flutter_tools/lib/src/commands/generate_localizations.dart

```diff
@@ -151,16 +151,14 @@ class GenerateLocalizationsCommand extends FlutterCommand {
     );
     argParser.addFlag(
       'synthetic-package',
-      defaultsTo: true,
+      defaultsTo: !featureFlags.isExplicitPackageDependenciesEnabled,
```

Comment:
Can you explain at a high level why `synthetic-package` and `--enable-explicit-dependencies` conflict in this way? I can imagine it's something to do with the synthetic nature of synthetic packages and the explicit nature of explicit dependencies, but I assume you can answer this question faster than I can interpret the test changes without this context :)

Reply:
Sure.

The simplest answer is "it was too hard", or rather, the `flutter_gen`-based path made assumptions of plugins that were difficult to change, and given the fact `flutter_gen` was a feature we generally regretted, it made sense to sunset the feature to unlock this one.

We could have deprecated synthetic packages first, but it would mean adding another quarter to the release process. I think if we had faster releases I would have probably kept the features totally separate and have already removed `package:flutter_gen`.

<https://docs.flutter.dev/release/breaking-changes/flutter-generate-i10n-source> for more.